device.go

```diff
@@ -166,11 +177,64 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
-func (device *BlockDevice) attach(h hypervisor) error {
+func makeBlockDevIDForHypervisor(deviceID string) string {
+	return fmt.Sprintf("drive-%s", deviceID)
```

Comment:
qemu 2.9 + q35 ID's must be <=31
would you mind reduce the length of the ID?

Reply:
the size limit was added in 2.9?
I'll change this, thanks

Reply:
@amshinde yep, ```node-name``` is truncated to 31 characters

Reply:
Yep, that's correct.  "node-name" is truncated at 31 characters but "id" for some reason is not.  This can lead to a subtle problem where the backend (which uses "node-name") and frontend (which uses "id") devices cannot be associated with each other if you specify a node-name of > 31 characters.